dask/blockwise.py

```diff
@@ -363,7 +410,33 @@ def __dask_distributed_pack__(
         dsk = (SubgraphCallable(dsk, self.output, tuple(keys2)),)
         dsk, dsk_unpacked_futures = unpack_remotedata(dsk, byte_keys=True)
 
-        func = dumps_function(dsk[0])
+        # Handle `io_deps` serialization. Assume each element
+        # is a `BlockwiseDep`-based object.
+        packed_io_deps = {}
+        _complex = False
+        for name, blockwise_dep in self.io_deps.items():
+            packed_io_deps[name] = {
+                "__module__": blockwise_dep.__module__,
+                "__name__": type(blockwise_dep).__name__,
+                "state": blockwise_dep.__dask_distributed_pack__(self.output_blocks),
+            }
+            # Mark this object as "complex" if the first element
+            # contains an inline task. This logic only captures
+            # BlockwiseDepDict-based dependencies.
+            # TODO: Remove this logic once single-pass serialization
+            # removes the need to treat nested tasks differently.
+            if hasattr(blockwise_dep, "mapping"):
+                _complex = _maybe_complex(next(iter(blockwise_dep.mapping.values())))
```

Comment:
Okay - It seems that the CSV problem with nested tasks had two causes.  First, nested tasks are not detected if they are hidden within a tuple.  Changing the `BlockwiseDep` object to return a `list` resolved this immediate problem. 

The second problem was that we are only checing if `self.concatenate` is true when choosing how to serialize functions and arguments during distributed execution.  Therefore, to resolve the CSV issue for distributed exectuion, we also need logic like the line above to check if we will ultimately need to call `to_serialize` on the entire task (rather than constructing a args/kwargs dict).  In the future, I am hoping that the single-pass serialization changes will remove the need for this check.  This assumption is the only reason I didn't attempt to bake the "complex task" logic into `BlockwiseDep` in a more rigorous way.